lms/djangoapps/courseware/access.py

```diff
@@ -552,18 +553,40 @@ def can_load():
         students to see modules.  If not, views should check the course, so we
         don't have to hit the enrollments table on every module load.
         """
-        response = (
-            _visible_to_nonstaff_users(descriptor)
-            and _has_group_access(descriptor, user, course_key)
-            and
-            (
-                _has_detached_class_tag(descriptor)
-                or _can_access_descriptor_with_start_date(user, descriptor, course_key)
+        has_staff_access = _has_staff_access_to_descriptor(user, descriptor, course_key)
+        # if the user has staff access, they can load the module so this code doesn't need to run
+        if not has_staff_access:
+            # only check milestones for sequentials, as gated content doesn't apply to other block types
+            if descriptor.location.block_type == 'sequential':
+                course_milestones = milestones_helpers.get_course_content_milestones(
+                    course_key,
+                    descriptor.location,
+                    'requires',
+                    user.id
+                )
+            else:
+                course_milestones = []
+
+            if descriptor.location in [m['content_id'] for m in course_milestones]:
+                debug("Deny: user has not completed all milestones for content")
+                milestones_check = ACCESS_DENIED
+            else:
+                milestones_check = ACCESS_GRANTED
+
+            response = (
+                _visible_to_nonstaff_users(descriptor) and
+                milestones_check and
+                _has_group_access(descriptor, user, course_key) and
+                (
+                    _has_detached_class_tag(descriptor)
+                    or _can_access_descriptor_with_start_date(user, descriptor, course_key)
+                )
             )
-        )
+        else:
+            response = True
```

Comment:
Being inside this if block implies `has_staff_access`, so I don't think we need to explicitly set `response` - `if (has_staff_access or response)` below will return `True` either way.